pyproject.toml

```diff
@@ -29,10 +29,10 @@ streamlit = "^1.30.0"
 watchdog = "^3.0.0"
 scipy = "^1.12.0"
 scikit-learn = "^1.4.0"
-typer = "^0.9.0"
+typer = ">=0.9.0,<1.0.0"
```

Comment:
Isn't this the same thing?

Reply:
It isn't, `^` is following semantic versioning, and it has different behaviour based on the leading `0`:

- `^1.0.0` means `>=1.0.0,<2.0.0`
- `^0.1.0` means `>=0.1.0,<0.2.0`
- `^0.0.1` means `>=0.0.1, <0.0.2`

But because with `typer`, so far we always do just `typer.run(main)`, I think we don't have to stress so much about the versions.

Reply:
ooo, I thought caret meant "up to but not including the next major version" but I see you're right (https://python-poetry.org/docs/dependency-specification/#caret-requirements)! Thanks for the lesson!